src/backend/src/app.py

```diff
@@ -0,0 +1,230 @@
+    print(f"Processing message: {message} with chat_id: {chat_id}")
+    try:
+        # Handle PII redaction if enabled
+        if PII_ENABLED:
+            print(f"Redacting PII for message: {message} with chat_id: {chat_id}")
+            message = pii_redacter.redact(
+                text=message,
+                id=chat_id,
+                cache=True
+            )
+
+        # Extract utterances
+        print(f"Extracting utterances from message: {message}")
+        utterances = extract_client.chat_completion(message)
+        print(f"Utterances: {utterances}")
+
+        if not isinstance(utterances, list):
+            try:
+                utterances = json.loads(utterances)
+            except JSONDecodeError:
+                logging.warning("Failed to parse utterances, possibly harmful content")
+                return ['I am unable to process this request.']
+
+        # Process each utterance
+        for utterance in utterances:
+            try:
+                # Reconstruct PII if needed
+                if PII_ENABLED:
+                    utterance = pii_redacter.reconstruct(
+                        text=utterance,
+                        id=chat_id,
+                        cache=True
+                    )
+
+                # Try semantic kernel orchestration first
+                orchestrator = app.state.orchestrator
+                response = await orchestrator.process_message(utterance)
+                
+                if isinstance(response, dict) and response.get("error"):
+                    # If semantic kernel fails, use fallback
+                    print(f"Semantic kernel failed, using fallback for: {utterance}")
+                    response = fallback_function(
+                        utterance,
+                        "en",  # Assuming English for simplicity, adjust as needed
+                        chat_id
+                    )
+                    responses.append(response)
+                else:
+                    responses.append(response)
+
+            except Exception as e:
+                logging.error(f"Error processing utterance: {e}")
+                responses.append("I encountered an error processing part of your message.")
+
+    except Exception as e:
+        logging.error(f"Error in message processing: {e}")
+        responses = ["I apologize, but I'm having trouble processing your request."]
+
+    finally:
+        # Clean up PII cache if enabled
+        if PII_ENABLED:
+            pii_redacter.remove(id=chat_id)
+
+    return responses
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    # Setup app
+    try:
+        logging.basicConfig(level=logging.WARNING)
+        print("Setting up Azure credentials and client...")
+        print(f"Using PROJECT_ENDPOINT: {PROJECT_ENDPOINT}")
+        print(f"Using MODEL_NAME: {MODEL_NAME}")
+        creds = DefaultAzureCredential(exclude_interactive_browser_credential=False)
+        await creds.__aenter__()
```

Comment:
do we need these `await item.__aeenter__()` guards? this seems like a bit of an anti-pattern (asking for private variables of an object)

If your intention is to do this with a context manager lets do that

https://www.geeksforgeeks.org/python/aenter-in-python/